marlowe-test/src/Data/Jsonable.hs

```diff
@@ -0,0 +1,126 @@
+-----------------------------------------------------------------------------
+--
+-- Module      :  $Headers
+-- License     :  Apache 2.0
+--
+-- Stability   :  Experimental
+-- Portability :  Portable
+--
+-- | Wrapping values that can be converted to and from JSON.
+--
+-----------------------------------------------------------------------------
+
+
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+
+module Data.Jsonable
+  ( -- * Types
+    Jsonable(..)
+  , JsonableType(..)
+  , KnownJsonable
+    -- * Conversion
+  , fromJsonable
+  , isKnownJson
+  , toJsonable
+    -- * Testing
+  , arbitraryJsonable
+  , checkRoundTripJsonable
+  , generateJsonable
+  , roundTripJsonable
+  ) where
+
+
+import Data.Aeson (FromJSON(..), Result(..), ToJSON(..), Value, fromJSON)
+import Data.Proxy (Proxy(..))
+import Test.Tasty.QuickCheck (Arbitrary(..), Gen)
+
+
+-- | A list of known types with `FromJSON`, `ToJSON`, and `Arbitrary` instances.
+type KnownJsonable = [JsonableType]
+
+
+-- | A type with `FromJSON`, `ToJSON`, and `Arbitrary` instances.
+data JsonableType = forall a. (ToJSON a, FromJSON a, Arbitrary a) => JsonableType {typeKey :: String, typeValue :: Proxy a}
```

Comment:
It might be too much to require Arbitrary for all the types, as we only require them for Token and Party. I see that you had to add arbitrary instances for Payment for example. No need to change it, just an observation